### ISSUE TITLE
Prove Optimized Code

### ISSUE THREAD

**Issue by sjudson** [nexus-xyz/nexus-zkvm#155]
In https://github.com/nexus-xyz/nexus-zkvm/pull/138 we set `opt-level = 0` for the default proving profile (`release-unoptimized`, as opposed to `opt-level = 3` for `release`). This was due to the appearance of PANICs when trying to prove some programs at higher optimization levels, such as https://github.com/nexus-xyz/nexus-zkvm/blob/main/examples/src/bin/galeshapley.rs. 

It would be good to explore the root causes of these PANICs, and try to improve the compilation pipeline so that we can successfully prove optimized programs (which should require fewer instructions and therefore be faster).

**Comment by yoichi-nexus** [comment]
I'm having difficulty reproducing this issue.

* I modified `Cargo.toml`
```
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -99,7 +99,7 @@ lto = true
 inherits = "release"

 [profile.release-unoptimized.package.example]
-opt-level = 0
+opt-level = 3
```
* ran `cargo clean`
* built `cargo-nexus` with `cargo build --release --package nexus-tools --bin cargo-nexus`
* and ran `galeshapley` with `examples % ../target/release/cargo-nexus nexus run --bin galeshapley` successfully.

**Comment by sjudson** [comment]
`cargo nexus run` always rebuilds/executes the code in debug mode:

```
*[main][~/Code/nexus/nexus-zkvm/examples]$ ../target/release/cargo-nexus nexus run --bin galeshapley
    Finished dev [unoptimized + debuginfo] target(s) in 0.48s
```

You need to execute the code either using the `nexus_core` interfaces or using `cargo nexus prove`, which will use a release(-unoptimized) build.

**Comment by sjudson** [comment]
Closed by #250
